tests/contrib/celery/test_integration.py

```diff
@@ -38,6 +38,8 @@ def fn_task():
         for result in results:
             assert result.get(timeout=self.ASYNC_GET_TIMEOUT) == 42
 
+        # Wait for all spans to finish
+        sleep(0.5)
         traces = self.pop_traces()
         assert len(traces) == (200 if self.ASYNC_USE_CELERY_FIXTURES else 100)
```

Comment:
This test is particularly flaky because it generares 100-200 spans while other celery tests produce only a handful of spans. We can reduce the number of spans by a factor of 10 and that might help reduce the frequency of test failures